src/java.base/share/classes/sun/security/ssl/SignatureScheme.java

```diff
@@ -375,10 +387,34 @@ static List<SignatureScheme> getSupportedAlgorithms(
             List<ProtocolVersion> activeProtocols) {
         List<SignatureScheme> supported = new LinkedList<>();
 
-        List<SignatureScheme> schemesToCheck =
-                config.signatureSchemes == null ?
-                    Arrays.asList(SignatureScheme.values()) :
-                    namesOfAvailable(config.signatureSchemes);
+        // If config.signatureSchemes is non-null then it means that
+        // it was defined by a system property.  Per
+        // SSLConfiguration.getCustomizedSignatureScheme() the list will
+        // only contain schemes that are in the enum.
+        // Otherwise, use the enum constants (converted to a list).
+        //
+        // Additional logic is added here to remove the ecdsa_brainpoolP512r1tls13_sha512
+        // signature scheme by default. We only want to make use of ecdsa_brainpoolP512r1tls13_sha512
+        // when explicitly set via system properties jdk.tls.client.SignatureSchemes or
+        // jdk.tls.server.SignatureSchemes.
+        List<SignatureScheme> schemesToCheck;
+        if (config.signatureSchemes != null) {
+            schemesToCheck = namesOfAvailable(config.signatureSchemes);
+        } else {
+            List<SignatureScheme> schemes = Arrays.asList(SignatureScheme.values());
+            schemesToCheck = new ArrayList<>(schemes.size());
```

Comment:
This can use the array directly here:
```java
            SignatureScheme[] schemes = SignatureScheme.values();
            schemesToCheck = new ArrayList<>(schemes.length);
```